### ISSUE TITLE
Double free or corruption in BlockedQueue test

### ISSUE THREAD

**Issue by marcusnaslund** [magic-lang/ooc-kean#1523]
Hadn't seen it before in the office, but on my home laptop I keep getting

```
BlockedQueue *** Error in `./Tests': double free or corruption (!prev): 0x0000000001313eb0 ***
```

pretty often, when running the concurrent tests.
This is a test not running properly on Windows, so I guess now we know why.

**Comment by marcusnaslund** [comment]
@niklasborwell Is this the problem you were talking about?

**Comment by niklasborwell** [comment]
It's hard to say since these kind of errors are so hard to debug, but it's the same kind of error. I encounter it randomly when running Vidhance.

**Comment by marcusnaslund** [comment]
Changing `BlockedQueue enqueue` to
```ooc
enqueue: override func (item: __onheap__ T) {
		this _mutex lock()
		this _backend enqueue(item)
		this _waitLock wake()
		this _mutex unlock()
	}
```
removes the problem on my laptop, as per http://stackoverflow.com/questions/1640389/pthreads-pthread-cond-signal-from-within-critical-section

Must be tested some more though.

**Comment by marcusnaslund** [comment]
Fixed by #1603
